contracts/ERC721Collateralizer.sol

```diff
@@ -0,0 +1,372 @@
+        require(agreementToCollateralizer[agreementId] != address(0));
+
+        // Ensure debt is in a state of default.
+        require(
+            termsContract.getExpectedRepaymentValue(
+                agreementId,
+                block.timestamp
+            ) > termsContract.getValueRepaidToDate(agreementId)
+        );
+
+        // Mark agreement's collateral as withdrawn by setting the agreement's
+        // collateralizer to 0x0.
+        delete agreementToCollateralizer[agreementId];
+
+        // Determine beneficiary of the collateral to be seized.
+        address beneficiary = debtRegistry.getBeneficiary(agreementId);
+
+        ERC721 collateralToken = ERC721(collateralTokenAddress);
+
+        // Transfer the collateral this contract was holding in escrow to beneficiary.
+        collateralToken.transferFrom(address(this), beneficiary, collateralTokenID);
+
+        // Log the seizure event.
+        CollateralSeized(
+            agreementId,
+            beneficiary,
+            collateralTokenAddress,
+            collateralTokenID
+        );
+    }
+
+    /**
+     * Adds an address to the list of agents authorized
+     * to invoke the `collateralize` function.
+     */
+    function addAuthorizedCollateralizeAgent(address agent)
+        public
+        onlyOwner
+    {
+        collateralizationPermissions.authorize(agent, CONTEXT);
+    }
+
+    /**
+     * Removes an address from the list of agents authorized
+     * to invoke the `collateralize` function.
+     */
+    function revokeCollateralizeAuthorization(address agent)
+        public
+        onlyOwner
+    {
+        collateralizationPermissions.revokeAuthorization(agent, CONTEXT);
+    }
+
+    /**
+    * Returns the list of agents authorized to invoke the 'collateralize' function.
+    */
+    function getAuthorizedCollateralizeAgents()
+        public
+        view
+        returns (address[])
+    {
+        return collateralizationPermissions.getAuthorizedAgents();
+    }
+
+    /**
+     * Unpacks collateralization-specific parameters from their tightly-packed
+     * representation in a terms contract parameter string.
+     */
+    function unpackCollateralParametersFromBytes(bytes32 parameters)
+        public
+        pure
+        returns (uint, uint)
+    {
+        bytes32 collateralTokenIndexShifted =
+            parameters & 0x0000000000000000000000000000000000000fffffffffffff00000000000000;
+        bytes32 tokenIdShifted =
+            parameters & 0x00000000000000000000000000000000000000000000000000ffffffffffffff;
+
+        // Shift the token index value 14 places to the right.
+        uint collateralTokenIndex = uint(collateralTokenIndexShifted) / 2 ** 56;
+
+        return (
+            collateralTokenIndex,
+            uint(tokenIdShifted)
+        );
+    }
+
+    function retrieveCollateralParameters(bytes32 agreementId)
+        internal
```

Comment:
There isn't necessarily any reason this can't be public -- would help with debugging stuff in the future.